shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -636,11 +622,29 @@ public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOE
             UUID writerId = new UUID(in.readLong(), in.readLong());
             long eventNumber = in.readLong();
             long expectedOffset = in.readLong();
-            Event event = Event.readFrom(in, length - 8 * 5);
+            Event event = readEvent(in, in.available());
             long requestId = in.available() >= Long.BYTES ? in.readLong() : -1L;
+            // All the available bytes should be read.
+            if (in.available() > 0) {
+                throw new InvalidMessageException("Was expecting zero length but found: " + in.available());
```

Comment:
This should not throw. It prevents us from adding new fields to this message in the future.

Reply:
At the same time, if we remove, then we could be missing an ill-formed message. I think we should be checking precisely the length according to version. If we want to relax the verification to be forward compatible, then we are opening ourselves to other problems. I'm in favor of supporting only backward compatibility on the server side, which means that we only accept messages with the same version of earlier, and we validate precisely the length of messages per version.

Reply:
@tkaitchuck @fpj I have improved the check to validate length. Does it make it any better now ?

Reply:
Our current policy is to allow additional data for forward compatibility: https://github.com/pravega/pravega/wiki/PDP-21:-Protocol-revisioning
We should stick with that for this PR.

Reply:
Yes, with this check the older server can throw an error if there is a newer version of `Conditional Append`. Removed this check.